container/variables.tf

```diff
@@ -6,12 +6,22 @@ variable "image_name" {
   description = "Name of the Docker image"
 }
 
-variable "container_port" {
-  description = "Port to expose in the container"
+variable "port_mapping" {
+  type = list(
+    object(
+      {
+        container_port = string,
+        host_port = string,
+        protocol = string
+      }
+    )
+  )
+  description = "A list of maps of ports to be deployed for the container"
+  default = null
 }
 
-variable "host_port" {
-  description = "Port to expose on the host"
+variable "ip_address" {
```

Comment:
I noticed that you had "ip_address" variable, but not sure where to use it.
if I have to guess will be on port assignments, right ?

Reply:
Yep, it means if you want to map it to a specific host ip address you can.

Default behaviour by the provider is to do it on all host IPs it has access to.